eole/predict/inference.py

```diff
@@ -674,8 +677,7 @@ def _decode_and_generate(
         else:
             src_pad_mask = None
         tgt_pad_mask = decoder_in.eq(self._tgt_pad_idx).unsqueeze(1)  # [B, 1, T_tgt]
-
-        emb = self.model.tgt_emb(decoder_in, step=step)
+        position_embeddings = self.model.rope.update(decoder_in.size(1), step=step)
```

Comment:
I find this a bit weird to have it in the generic generator. In theory we could have some models that do not have position embeddings. (The no-op is there to cover this case, but it's a bit unclear.)

Reply:
making things 100% callable is way better for torch.compile (vs "if" everywhere, same logic as removing all of these in transformer_decoder / transformer_lm_decoder)